### PR TITLE
boards: arm: fk7b0m1_vbt6: add support to external NOR Flash

### PR DESCRIPTION
**Pull request by CharlesDias**
Updates the DTS file by adding support for Quad SPI and SPI NOR Flash.

This update has been validated via **samples/drivers/spi_flash** and **samples/drivers/flash_shell/**.

```
*** Booting Zephyr OS build zephyr-v3.5.0-4931-ga37bd8e7ba34 ***

spi-nor-flash@0 SPI flash testing
==========================

Perform test on single sector
Test 1: Flash erase
Flash erase succeeded!

Test 2: Flash write
Attempting to write 4 bytes
Data read matches data written. Good!!

Perform test on multiple consequtive sectors
Test 1: Flash erase
Flash erase succeeded!

Test 2: Flash write
Attempting to write 4 bytes at offset 0xff000
Data read matches data written. Good!!
Attempting to write 4 bytes at offset 0x100000
Data read matches data written. Good!!
```

```
Flash shell sample[00:00:00.100,000] <inf> flash_stm32_ospi: OSPI flash config is SPI|DUAL|QUAD / STR
[00:00:00.100,000] <inf> flash_stm32_ospi: Read SFDP from octoFlash
[00:00:00.100,000] <inf> flash_stm32_ospi: Read SFDP from octoFlash
*** Booting Zephyr OS build zephyr-v3.5.0-4932-gb3541b2e175e ***
uart:~$ flash page_info qspi-nor-flash@0 0
Page for address 0x0:
start offset: 0x0
size: 4096
index: 0
uart:~$ flash erase qspi-nor-flash@0 0x1000
Erase success.
[00:00:57.477,000] <inf> flash_stm32_ospi: Sector/Block Erase
[00:00:57.477,000] <inf> flash_stm32_ospi: Sector/Block Erase addr 0x1000, asize 0x2000 amode0
uart:~$ flash write qspi-nor-flash@0 0x1000 0x12345678 0x9abcdef0
Write OK.
Verified.
uart:~$ flash read qspi-nor-flash@0 0x1000 0x10
00001000: 78 56 34 12 f0 de bc 9a  ff ff ff ff ff ff ff ff |xV4..... ........|

uart:~$ flash page_info spi-nor-flash@0 0
Page for address 0x0:
start offset: 0x0
size: 65536
index: 0
uart:~$ flash erase spi-nor-flash@0 0x1000
Erase success.
uart:~$ flash write spi-nor-flash@0 0x1000 0x12345678 0x9abcdef0
Write OK.
Verified.
uart:~$ flash read spi-nor-flash@0 0x1000 0x10
00001000: 78 56 34 12 f0 de bc 9a  ff ff ff ff ff ff ff ff |xV4..... ........|

uart:~$ 
```

### CODE REVIEW

**Comment by CharlesDias** [comment]
Dear @kartben, 

Could you help me with this PR? Thanks!

**Comment by CharlesDias** [comment]
Solved conflicts.

**Comment by benediktibk** [comment]
Looks like nothing covers boards/fanke in the maintainers file. As this board has a STM32 on it I was so free to add the ST people.

File: boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts [hunk 1]
```diff
@@ -75,6 +76,61 @@
...
+		     &octospim_p1_io2_pe2 &octospim_p1_io3_pd13>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	/* Winbond external flash */
+	w25q64jvssiq_qspi: qspi-nor-flash@0 {
+		compatible = "st,stm32-ospi-nor";
+		reg = <0>;
+		ospi-max-frequency = <DT_FREQ_M(133)>;
+		size = <DT_SIZE_M(64)>; /* 64 Mbits */
```
Comment (erwango) [inline boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts:91]: `size`  is now deprecated, see https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/releases/migration-guide-3.6.rst#flash

File: boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts [hunk 2]
```diff
@@ -75,6 +76,61 @@
...
+&spi6 {
+	pinctrl-0 = <&spi6_sck_pb3 &spi6_miso_pb4 &spi6_mosi_pb5>;
+	cs-gpios = <&gpioa 15 GPIO_ACTIVE_LOW>;
+	pinctrl-names = "default";
+	status = "okay";
+	w25q64jvssiq_spi: spi-nor-flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+		size = <DT_SIZE_M(64)>; /* 64 Mbits */
```
Comment (erwango) [inline boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts:116]: Ditto on `size` prop

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts [hunk 3]
```diff
@@ -75,6 +76,61 @@
...
+
+&spi6 {
+	pinctrl-0 = <&spi6_sck_pb3 &spi6_miso_pb4 &spi6_mosi_pb5>;
+	cs-gpios = <&gpioa 15 GPIO_ACTIVE_LOW>;
+	pinctrl-names = "default";
+	status = "okay";
+	w25q64jvssiq_spi: spi-nor-flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
```
Comment (GeorgeCGV) [inline boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts:115]:
```suggestion
		spi-max-frequency = <DT_FREQ_M(40)>;
```

File: boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts (hunk 2 above)
Comment (CharlesDias) [reply boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts:116]: @erwango, If a remove `size` and change to `reg = <0 DT_SIZE_M(8)>;` I got the error message
`error: static assertion failed: "jedec,spi-nor size required for non-runtime SFDP page layout"`

I'm up-to-date with the main branch (hash commit c6e482714f2387c).

Comment (erwango) [reply boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts:116]: ok, my bad. This is indeed a different compatible.

File: boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts (hunk 3 above)
Comment (CharlesDias) [reply boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts]: Done. Thanks!

File: boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts (hunk 1 above)
Comment (CharlesDias) [reply boards/fanke/fk7b0m1_vbt6/fk7b0m1_vbt6.dts]: Done. Thanks!

**Review by GeorgeCGV — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**